extensions/mssql/src/ui/scriptableDialogBase.ts

```diff
@@ -63,13 +63,16 @@ export abstract class ScriptableDialogBase<OptionsType extends ScriptableDialogO
 	protected abstract get isDirty(): boolean;
 
 	protected override onFormFieldChange(): void {
-		this._scriptButton.enabled = this.isDirty;
+		this.updateScriptButtonState();
 		this.dialogObject.okButton.enabled = this.isDirty;
```

Comment:
Is this correct? Shouldn't it also be checking the valid state? I'm assuming that's done by the core - but by setting it here wouldn't that be overriding it?

In fact, wouldn't it be better/easier to just have the valid state of the dialog itself use isDirty to determine validity? If it's not dirty then there's nothing to do so it makes sense to have it be "not valid" I think.

(you'd still need to manually update the script enabled state, but at that point it could just be simply `= this.modelView.valid`)